recipe/meta.yaml

```diff
@@ -1,113 +1,85 @@
-{% set version = "2.6.0" %}
+{% set name = "keras" %}
+{% set version = "3.0.5" %}
 
 package:
-  name: keras_split
+  name: keras
   version: {{ version }}
 
 source:
-  - url: https://pypi.io/packages/py2.py3/k/keras/keras-{{ version }}-py2.py3-none-any.whl
-    sha256: 504af5656a9829fe803ce48a8580ef16916e89906aceddad9e098614269437e7
-  - url: https://raw.githubusercontent.com/keras-team/keras/v{{ version }}/LICENSE
-    sha256: ebc07b279109d4bd5e58fe06f54e8f8aee8ba1024b77b222bc04214ae6f56348
+  url: https://github.com/keras-team/keras/archive/refs/tags/v{{ version }}.tar.gz
+  sha256: 0e23205f2d8f434ded17ca9efb116cdb7b6d19736e576456c28aea685a7f195f
 
 build:
-  noarch: python
+  script_env:
+    # The default backend is tensorflow, but for this version of keras
+    # we do not have the correct version. We update keras and would like
+    # to offer at least a working backend: torch.
+    - KERAS_BACKEND=torch
```

Comment:
It is a runtime variable as far as I can see. 

If it's not specified, the fallback is the default `tensorflow`, as expected.

I set it in the `build` section to `torch`, so that the first `import keras` won't fail while trying to import `tensorflow`.

Reply:
does it import keras somewhere as part of the build process (i.e. when you run pip install?)

Reply:
not in `setup.py` 

but in `pip_build.py` which is used to create a .whl on their ci:
```
        from keras.src.version import __version__  # noqa: E402
```
https://github.com/keras-team/keras/blob/b36f03731954037ea3b8a0f334368ceedad364b8/pip_build.py#L88

I don't think this will be a problem.

Reply:
yeah, I don't think it's a problem to leave it in, but I don't think it does anything either. If that's right, could be misleading for future maintainers.

Reply:
Actually I would try building this without this variable.
Because if `keras` is not imported during the build step, setting this variable is pretty much useless.
If the build fails without this variable having been set, then yes, include it by all means.

I think where this variable should be set is the activation script.

Reply:
yes, correct, but my understanding is that we want to have PyTorch as the only backend available for now, until `tensorflow` is updated.
In this case, `torch` should also be a `run` dependency of course.

Reply:
We support both pytorch and jax. Pytorch should not be a runtime dependency.

Reply:
I thought we had come to a decision that we weren't swapping the default backend to pytorch, because it would change behaviour for users? rather that we were telling snowflake to configure the backend themselves?

Reply:
and connectedly, all backends should be run_constrained because they're optional. unless it's a special build. (a package called `keras-tensorflow` maybe)

Reply:
I also wonder what would be the behaviour of the keras that tensorflow depends on if pytorch is in the same environment

Reply:
sorry for the somewhat rant-y tone - for background, keras is a very popular AI package and this has the potential to have a significant impact

Reply:
well, we can always re-release keras with the default backend changed and the build number bumped when we release tensorflow, I guess, to stop such breakage. Seems unnecessary though.

Reply:
thank you for helping here @danpetry , I just need `KERAS_BACKEND=torch` for the first `import keras` in the test section, and not a build time.

The default backend is still `tensorflow`, but I can't run tests with that, I need to run them on `torch`.

Do we have a way to set this variable just right before the tests start? Otherwise, I will remove the `imports` section and use a `KERAS_BACKEND=torch python -c "import keras;"` command under `commands`.

Reply:
the test commands are just bash commands, so you can `export KERAS_BACKEND=torch` before any of the tests

Reply:
moved `tests/imports` section to the `test/commands` section

Reply:
ah I see. right yeah